README.md

```diff
@@ -1,73 +1,320 @@
+
+### auto
+
+What operation, if any, should be done automatically at IPsec startup. Currently-accepted values are **add**, **ondemand**, **start**, and **ignore** (also the default) (signifying no automatic startup operation).
+
+### hosts
+Each key in this dictionary is the unique name of a host. For each host key in this dictionary, the following host-specific parameters can be specified.  
+ 
+| Parameter            | Description                                                                                       | Type        | Required | Default                 | Libreswan Equivalent |
+|----------------------|---------------------------------------------------------------------------------------------------|:-----------:|:--------:|-------------------------|:--------------------:|
+| host                 | A host/IP to use for setting up a vpn connection. Only specified when overriding the name of hosts used by Ansible for SSH. | str | no   | - | left/right           |
+| public_key_src       | A path to a file on the controller host containing the public key used by this host for public key authentication without certificates.     | str    | no       | See [public_key fields](#public_key)| leftrsasigkey/rightrsasigkey |
+| public_key_content   | Contains the public key used by this host for public key authentication without certificates.                        | str         | no       | -                       | leftrsasigkey/rightrsasigkey |
+| cert_name            | Defines the certificate nickname of this host's certificate in the NSS database.                  | str         | no       | -                       | leftcert/rightcert   |
+| external             | Indicates that the associated host is external to the inventory (e.g. not managed by our inventory).    | bool        | no       | False                   | -                    |
+| subnets              | Defines a list of the subnets that should be available via the VPN connection.                    | list        | no       | -                       | leftsubnets/rightsubnets |
+
+#### public_key
+
+`public_key_src` specifies a path to a file on the controller host containing the public key used by this host for public key authentication without certificates. Otherwise, the user can directly specify the public key for this host by populating `public_key_content`. `public_key_content` can also accept a CKAID or nickname for a public key in the NSS database.
+
+Note that `public_key_src` and `public_key_content` may also be specified as host variables in the Ansible `all.hosts` section. The variable names in this case will be `vpn_public_key_src` and `vpn_public_key_content`. See [this example](#defining-public-keys-at-different-levels).
+
+**To be added later**: if neither `public_key_src` nor `public_key_content` is populated, the role will generate key pairs for each host.
```

Comment:
What would happen if the role is rerun against the same set of host? Would it regenerate the key pairs, or keep the existing?

Reply:
Good question. As the role is designed now, unless the user references an existing public key using the `public_key_src` or `public_key_content` fields, the role would regenerate key pairs.

Reply:
@richm this kind of brings up the general issue of idempotency in running this role. I have been assuming that if a vpn tunnel already exists between two hosts, running the role to "set up" the connection will update the existing configuration. Is this how it should work?

Reply:
@mprovenc the `template` module is idempotent - but the method by which keys are generated should also be idempotent - the role should check if a key already exists, and do not generate one if it already exists - this may mean we need to add some sort of parameter like `vpn_regen_keys: true` in order to force regeneration